### PR TITLE
Add keyword to match on 'raw' TLS certificates

### PR DESCRIPTION
**Pull request by thus**
Add new keyword `tls.cert` to match on the 'raw' TLS certificate. This keyword matches on the first certificate in the certificate chain.

Example:
```alert tls any any -> any any (msg:"test tls.cert"; tls.cert; content:"|01 02 03 04|"; sid:1;)```

Updates:
- Fixed changes requested in #3851 
- Changed the code so it inspects every certificate in the chain

https://redmine.openinfosecfoundation.org/issues/2670

Prscript:
- PR thus-pcap: https://buildbot.openinfosecfoundation.org/builders/thus-pcap/builds/185
- PR thus: https://buildbot.openinfosecfoundation.org/builders/thus/builds/185

### CODE REVIEW

**Comment by thus** [comment]
https://github.com/OISF/suricata-verify/pull/56

File: doc/userguide/rules/tls-keywords.rst [hunk 1]
```diff
@@ -136,6 +136,21 @@ Usage::
 
   tls_cert_valid;
 
+tls.cert
+--------
+
+Do a "raw" match on the TLS certificate. This keyword only matches on the
+first certificate in the certificate chain.
```
Comment (victorjulien) [inline doc/userguide/rules/tls-keywords.rst]: no it doesn't, it matches on all of them... right?

File: src/detect-tls-cert.c [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    const DetectEngineTransforms *transforms;
+} PrefilterMpmTlsCert;
+
+/**
+ * \brief Registration function for keyword: tls.cert
+ */
+void DetectTlsCertRegister(void)
+{
+    sigmatch_table[DETECT_AL_TLS_CERT].name = "tls.cert";
+    sigmatch_table[DETECT_AL_TLS_CERT].desc = "content modifier to match the TLS certificate buffer";
```
Comment (victorjulien) [inline src/detect-tls-cert.c]: sticky buffer
Also, should this mention that we only inspect certs the server provides?

**Comment by thus** [comment]
> I wonder if this should be tls.server_certs or something? Or do you not see a case for client certs being added later?

It would be cool to add client certificates too. I haven't really thought about that in a while, but it would probably not be a huge job to add (famous last words).

If I do that, would I then be allowed to keep the name "tls.cert"? ;)

File: doc/userguide/rules/tls-keywords.rst (hunk 1 above)
Comment (thus) [reply doc/userguide/rules/tls-keywords.rst]: Yeah. I should probably fix that!

**Comment by thus** [comment]
Replaced by #3857

**Pull request closed without merging**